src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesArgs.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using NuGet.Common;
+using NuGet.Configuration;
+
+namespace NuGet.Commands
+{
+    public class SourcesArgs
+    {
+        public delegate void Log(string message);
+
+        public ISettings Settings { get; }
+        public IPackageSourceProvider SourceProvider { get; }
+        public SourcesAction Action { get; }
+        public string Name { get; }
+        public string Source { get; }
+        public string Username { get; }
+        public string Password { get; }
+        public bool StorePasswordInClearText { get; }
+        public string ValidAuthenticationTypes { get; }
+        public SourcesListFormat Format { get; }
+        public bool Interactive { get; }
+        public string ConfigFile { get; }
+        private bool IsQuiet { get; }
+        public Log LogError { get; }
+        public Log LogMinimal { get; }
+        public Log LogQuiet { get; }
+
+        public void LogQuietImplementation(string data)
```

Comment:
nikolche and i discussed. 
i need to make LogQuiet use logger.LogMinimal.
i need to make LogMinimal use logger.LogMinimal if !IsQuiet
building and testing now.

Reply:
tested this, believe we are all good now.